system/jlib/jmutex.hpp

```diff
@@ -411,7 +413,17 @@ class jlib_decl  SpinLock
             return;
         }
         while (unlikely(!atomic_acquire(&value)))
-            ThreadYield();
+        {
+            unsigned i = 0;
```

Comment:
I wonder if this code should be in a non-inline function.  That will reduce the size of the generated code, and possibly increase the number of registers available to the place this is inlined.